pkg/trace/obfuscate/sql_tokenizer.go

```diff
@@ -480,60 +493,68 @@ func (tkn *SQLTokenizer) scanString(delim rune, kind TokenKind) (TokenKind, []by
 }
 
 func (tkn *SQLTokenizer) scanCommentType1(prefix string) (TokenKind, []byte) {
-	buffer := &bytes.Buffer{}
-	buffer.WriteString(prefix)
 	for tkn.lastChar != EOFChar {
 		if tkn.lastChar == '\n' {
-			tkn.consumeNext(buffer)
+			tkn.next()
 			break
 		}
-		tkn.consumeNext(buffer)
+		tkn.next()
 	}
-	return Comment, buffer.Bytes()
+	return Comment, tkn.grab()
 }
 
 func (tkn *SQLTokenizer) scanCommentType2() (TokenKind, []byte) {
-	buffer := &bytes.Buffer{}
-	buffer.WriteString("/*")
 	for {
 		if tkn.lastChar == '*' {
-			tkn.consumeNext(buffer)
+			tkn.next()
 			if tkn.lastChar == '/' {
-				tkn.consumeNext(buffer)
+				tkn.next()
 				break
 			}
 			continue
 		}
 		if tkn.lastChar == EOFChar {
 			tkn.setErr("unexpected EOF in comment")
-			return LexError, buffer.Bytes()
+			return LexError, tkn.grab()
 		}
-		tkn.consumeNext(buffer)
+		tkn.next()
 	}
-	return Comment, buffer.Bytes()
+	return Comment, tkn.grab()
 }
 
-func (tkn *SQLTokenizer) consumeNext(buffer *bytes.Buffer) {
-	if tkn.lastChar == EOFChar {
-		// This should never happen.
-		panic("unexpected EOF")
+// Next sets tkn.lastChar to the next rune at tkn.buf[tkn.index], and advances the tkn.pos and
+// tkn.index by its size. If next() is unable to scan a character, tkn.lastChar is set to EOFChar.
```

Comment:
The documentation is a bit too "code like". Maybe it should just say
```go
// next advances the tokenizer to the next rune. If this is not possible, tkn.lastChar will
// be set to EOFChar.
``` 
The technical details are in the code and the scope of the fields are in the structure definition comments.

Since this is called a "tokenizer", it's easy to think that `next` takes us to the next token, so maybe `nextRune` might be better, or `advance`

Reply:
👍  I will simplify the comment.
I like `advance` as a name.